.release-notes/4505.md

```diff
@@ -0,0 +1,10 @@
+## Fix: Segmentation fault when ponyc compiles this code
+
+Fix an issue caused by passing a wrong expression to `consume`.
+
+With this change, instead of the compiler generating a
+segmentation fault, it generates the error message: "You cannot
+consume an expression that is not local. Specifically, you can
+only consume a local variable (a single lowercase identifier, without
+a dot) or a this field (this followed by a dot and a single lowercase
+identifier)".
```

Comment:
```suggestion
## Fix compiler segmentation fault

Previously the following code would cause the compiler to crash. Now it will display a helpful error message:

```pony
struct FFIBytes
  var ptr: Pointer[U8 val] = Pointer[U8].create()
  var length: USize = 0

  fun iso string(): String val =>
    recover String.from_cpointer(consume FFIBytes.ptr, consume FFIBytes.length) end

actor Main
  new create(env: Env) =>
    env.out.print("nothing to see here")
```
```